src/main.ts

```diff
@@ -21,6 +21,8 @@ export async function main(): Promise<Outputs> {
     const deploymentMode = getInput('deploymentMode')
     const deploymentName = getInput('deploymentName')
     const parameters = getInput('parameters')
+    const managementGroupId = getInput('managementGroupId')
+    const validationOnly = getInput('validationOnly') == 'true';
 
     // change the subscription context
     info("Changing subscription context...")
```

Comment:
setting subscription context should not be done management group scope as subscription id would be an optional input

Reply:
Also make it an optional input in action.yml